core/src/apps/management/apply_settings.py

```diff
@@ -124,3 +131,18 @@ async def require_confirm_change_autolock_delay(ctx, delay_ms):
     text.normal("Do you really want to", "auto-lock your device", "after")
     text.bold("{}?".format(format_duration_ms(delay_ms)))
     await require_confirm(ctx, text, ButtonRequestType.ProtectCall)
+
+
+async def require_confirm_unsafe_prompts(ctx, allow: bool) -> None:
+    if allow:
+        text = Text("Unsafe prompts", ui.ICON_WIPE)
+        text.normal(
+            "Trezor will allow you to", "confirm actions which", "might be dangerous."
+        )
+        text.br_half()
+        text.bold("Allow unsafe prompts?")
+        await require_hold_to_confirm(ctx, text, ButtonRequestType.ProtectCall)
+    else:
+        text = Text("Unsafe prompts", ui.ICON_CONFIG)
+        text.normal("Do you really want to", "disable unsafe prompts?")
```

Comment:
It would nice to add something more descriptive here as well. Maybe we could add "Trezor will be more strict again.". But since the user was informed before when turning it on, I guess it is not needed.

Reply:
I'd leave that for the redesign/rewording? Maybe all screens will get descriptive texts then.

Reply:
Sounds good :+1: